<launch>
  <node pkg="nodelet" type="nodelet" name="threadpool"  args="manager" output="screen">
    <param name="num_worker_threads" value="10" />
  </node>
  
  <node pkg="resource_management" type="resource_manager" name="resource_manager" output="screen">
    <rosparam>
      num_request_threads: 10
      resources:
        usb_bandwidth:
          available: 480
          max: 480
          min: 0
    </rosparam>
  </node>
  
  <include file="$(find v4l2_cam)/cam_nodelet.launch">
    <arg name="device_path" value="/dev/video1"/>
    <arg name="camera_name" value="pseye_000_wide"/>
    <arg name="name" value="camera0"/>
    <arg name="manager_name" value="threadpool"/>
    <arg name="resource_name" value="resource_manager"/>
  </include>
  
  <include file="$(find v4l2_cam)/cam_nodelet.launch">
    <arg name="device_path" value="/dev/video2"/>
    <arg name="camera_name" value="pseye_001_wide"/>
    <arg name="name" value="camera1"/>
    <arg name="manager_name" value="threadpool"/>
    <arg name="resource_name" value="resource_manager"/>
  </include>
  
  <node pkg="nodelet" type="nodelet" name="detector"
        args="load atags/detector_nodelet threadpool" output="screen">
    <rosparam>
      image_sources: [camera0/image_raw]
      tag_family: 36h11
    </rosparam>
    <remap from="detections" to="tag_detections"/>
  </node>
  
  <node pkg="v4l2_cam" type="camera_manager" name="camera_manager" output="screen">
    <rosparam>
      cameras: [camera0, camera1]
    </rosparam>
  </node>
  
  <node pkg="manycal" type="corresponder_node" name="corresponder" output="screen">
    <remap from="source" to="tag_detections"/>
    <rosparam param="camera_manager">camera_manager</rosparam>
  </node>
  
</launch>
